modules/build/src/main/scala/scala/build/internal/markdown/MarkdownCodeBlock.scala

```diff
@@ -0,0 +1,73 @@
+package scala.build.internal.markdown
+
+import scala.collection.mutable
+import scala.jdk.CollectionConverters._
+
+/** Representation for a (closed) code block contained in Markdown
+  *
+  * @param info
+  *   a list of tags tied to a given code block
+  * @param body
+  *   the code block content
+  * @param startLine
+  *   starting line on which the code block was defined (excluding backticks)
+  * @param endLine
+  *   end line on which the code block was closed (excluding backticks)
+  */
+case class MarkdownCodeBlock(
+  info: Seq[String],
+  body: String,
+  startLine: Int,
+  endLine: Int
+) {
+
+  /** @return
+    *   `true` if this snippet should be ignored, `false` otherwise
+    */
+  def shouldIgnore: Boolean = info.head != "scala" || info.contains("ignore")
+
+  /** @return
+    *   `true` if this snippet should have its scope reset, `false` otherwise
+    */
+  def resetScope: Boolean = info.contains("reset")
+
+  /** @return
+    *   `true` if this snippet is a test snippet, `false` otherwise
+    */
+  def isTest: Boolean = info.contains("test")
+
+  /** @return
+    *   `true` if this snippet is a raw snippet, `false` otherwise
+    */
+  def isRaw: Boolean = info.contains("raw")
+}
+
+object MarkdownCodeBlock {
+
+  /** Finds all code snippets in given input
+    *
+    * @param md
+    *   Markdown file in a `String` format
+    * @return
+    *   list of all found snippets
+    */
+  def findCodeBlocks(md: String): Seq[MarkdownCodeBlock] = {
+    val lines = md
+      .lines()
+      .toList
+      .asScala
+    lines
+      .zipWithIndex
+      .foldLeft((0, Seq.empty[MarkdownCodeBlock], Option.empty[MarkdownOpenFence])) {
```

Comment:
It would be better, if you move logic from `foldLeft` to tail recursive function.

Reply:
done